### DISCUSSION TITLE
[Discussion] Vendor (Microsoft) Charts (Sunburst, TreeMap etc.)

### DISCUSSION THREAD

**Discussion by ghost** [gitbrent/PptxGenJS#1115] [General]
Hi @gitbrent and others,

we have a customer-driven use-case to export sunbursts charts, which is missing in pptxgenjs. @incowiahpe (one of my co-workers) developed a prototype (https://github.com/incowiahpe/PptxGenJS) to test out if its possible, which it is. :)

During development of the prototype we got to the conclusion that extending "addChart(...)" is not a good idea, because sunburst charts are not part of OOXML, but instead specific to a... lets say vendor-extension from Microsoft and therefore its not possible to ensure compatibility across all presentation programs that understand OOXML, but most likely only to MS PowerPoint 2016 or later.

We've written down our idea - **which we like to discuss here** - in a pdf with (hopefully) all relevant information for the topic.
[sunburst_implementation_in_pptxgenjs.pdf](https://github.com/gitbrent/PptxGenJS/files/6778287/sunburst_implementation_in_pptxgenjs.pdf)

At the moment, our goal is to have the option to generate slices with sunburst charts, but there is also an upcoming use-case for a TreeMap (customer-driven, so I can't promise). Nevertheless, we are commited to implement it for pptxgenjs. We just need some feedback to get started
. :)

What do you think?

Best regards

**Comment by ghost** [comment]
#961 @rejochandran You may want to join the discussion? :)

**Comment by gitbrent** [comment]
Hi @CroniD 

I appreciate the feedback and chart implementation.

This seems more like something you create forks for. I'm not inclined to add support for non-OOXML charts especially if your org is the only one requesting it...

**Comment by ghost** [comment]
Edit: Changed writing to be less passive-aggressive. Sry, wrote it after waking up and before any coffee. :-/

Hi @gitbrent 

Thank you your response. I don't think we are the only ones. There is #493 at least, where you requested help (I assume because of the tag)? Well, its an old one. Nevertheless, the request for such charts will popup from time to time again, I believe. Not a strong reason to change your mind, I must admit.

But, what is your concern in adding support for vendor-specific extensions?

**Comment by veerasaraha** [comment]
Hi, Do you have any update on this adding this chart type? It would be a great add!
